openedx/core/djangoapps/user_api/accounts/tests/test_views.py

```diff
@@ -403,6 +403,11 @@ def test_patch_account(self, field, value, fails_validation_value=None, develope
         Test the behavior of patch, when using the correct content_type.
         """
         client = self.login_client("client", "user")
+
+        # Ensure the user has reasonable settings (including birth_year, so
+        # account_privacy works as expected)
+        self.create_mock_profile(self.user)
```

Comment:
Ah.  It was needed because if there is no birth year specified, the api treats the user as a minor, and returns "private" for the account_visibility field, regardless of what was submitted.  I can update this to only set the birth year field, or even to only set that field if we're testing account visibility.